go/parquet/internal/encoding/encoding_test.go

```diff
@@ -406,6 +406,17 @@ func (b *BaseEncodingTestSuite) TestDeltaByteArrayRoundTrip() {
 	}
 }
 
+func (b *BaseEncodingTestSuite) TestByteStreamSplitRoundTrip() {
+	b.initData(10000, 1)
+
+	switch b.typ {
+	case reflect.TypeOf(float32(0)), reflect.TypeOf(float64(0)), reflect.TypeOf(int32(0)), reflect.TypeOf(int64(0)), reflect.TypeOf(parquet.FixedLenByteArray{}):
```

Comment:
would you mind add a test case for `Put` and read with smaller batch-size? For example, the data-count is 10000, we can `Put(2500 values)` for 4 times and flush it. During read, we also read with smaller stride?

Reply:
@mapleFU I added a test that encodes/decodes in multiple chunks which surfaced a bug in how the encoder/decoder state was being handled between chunks. I've pushed up the fix, thanks for this feedback.

Reply:
Aha, this is great